packages/docs/content/docs/tutorials/supabase-mcp-server.mdx

```diff
@@ -0,0 +1,584 @@
+---
+title: 'Building a Supabase MCP Server for Natural Language SQL Queries in Pochi'
+---
+
+# Building a Supabase MCP Server for Natural Language SQL Queries in Pochi
+
+Often, when working with teams to debug code or build features, you need to look up data from your database. This may include checking if a recent migration seeded correctly, inspecting a particular table schema, or verifying specific column values. Normally, this means breaking your flow, opening the SQL client, writing queries by hand, or involving a co-worker.
+
+This tutorial is designed to help you avoid that. We’ll build a system where you can prompt Pochi right inside your code editor to answer all your database related queries using natural language prompts. For example: “Give me the table schema of the ‘sales_report’ table.”
+
+### Prerequisites
+
+Before we start, you’ll need the following: 
+
+- **[Install Pochi](https://docs.getpochi.com/#installation)** 
+- **[Setup a Supabase Project](https://supabase.com/dashboard/)**  
+- **[OpenAI API Key](https://platform.openai.com/)**  (You can use any other AI model if preferred.)
+
+
+## What is MCP (Model Context Protocol)?
+
+MCP (Model Context Protocol) is a lightweight, flexible interface that allows large language models (LLMs) like those used in Pochi to connect and interact with external tools, APIs, or databases. In this tutorial, MCP acts as the bridge between your natural language queries and the Supabase database, enabling Pochi to translate plain English questions into SQL queries using OpenAI, execute them, and even generate visualizations using Vega Charts.
+
+![Architectural Diagram](./images/mcp-server-supabase-diagram.png)
+
+We’re going to use Supabase database and limit to read only queries, but you can always extend to perform CRUD operations too. Using MCP servers makes it easy to extend AI assistants with custom functionality and is a perfect way to keep your focus on coding while still getting the data context you need.
+
+### Self host or use the managed service
+
+Some teams may prefer to run Pochi entirely on their own infrastructure rather than rely on the managed cloud service. Pochi provides you with both options. You can either self-host which means that the chat interface, orchestration, and MCP connections all run locally or on your servers. This is ideal if you require stricter compliance, private networking, or better cost control when integrating with internal systems.
+
```

Comment:
This section about self-hosting vs. managed services isn't directly relevant to the tutorial's goal of building a local MCP server. To keep the content focused and avoid potential confusion, it would be best to remove this part.